src/python/tests/test_libs/test_utils.py

```diff
@@ -173,6 +173,19 @@ def slow(func):
                              func)
 
 
+def android_device_required(func):
+  """Skip Android-specific tests if we cannot run them."""
+  reason = None
+  if not environment.get_value('INTEGRATION'):
+    reason = 'Integration tests are not enabled.'
+  elif environment.platform() != 'LINUX':
+    reason = 'Android device tests can only run on a Linux host.'
+  if not environment.get_value('ANDROID_SERIAL'):
```

Comment:
Should this be elif OR maybe different order ?

Reply:
Not using elif was unintended but this should probably be first as well. Fixed both.